README.md

```diff
@@ -0,0 +1,237 @@
+# Proposal: Container Core
+
+> A draft model for a test-framework-agnostic Docker Container abstraction (for Testcontainers 2.0)
+
+Author: [Richard North](mailto:rich.north@gmail.com)
+
+Date: October 2017
+
+
+
+This repository contains a skeleton structure for a new library, Container Core, which is documented below as a proposal. 
+
+
+
+## Goals
+
+* **Continue to provide an object abstraction for Docker containers**: This seems to have proven to be a popular way to interace with Docker from Java
+* **Allow Testcontainers to support test frameworks other than JUnit 4**: JUnit 5, Spock and TestNG would be the likely candidates
+* **Expose a reasonably simple API that emphasises discoverability of features and ease of use**: Again, the fluent setter pattern has proven to work well
+* **Allow us to clean up the Testcontainers API**: for example:
+  * making sure all fluent setters are consistent
+  * adding a separation between definition-time concerns and 'run-time' concerns
+  * add a plugin model to allow *composition* to be used in addition to *inheritance*
+* **Continue to support specialized container types via specialized classes** 
```

Comment:
Since we already started the discussion in Slack - we don't want to have Docker-Compose support as part of containercore, correct?